### PR TITLE
test(panel): Updated positioning tests with a custom matcher.

### PR DESCRIPTION
**Pull request by ErinCoughlan**
This reduces some flakiness around different viewports and rounding values with getBoundingClientRect.

@KarenParker @jelbourn @gmoothart @DerekLouie Please review.

### CODE REVIEW

File: src/components/panel/panel.spec.js [hunk 1]
```diff
@@ -31,6 +31,41 @@ describe('$mdPanel', function() {
...
+       * used for testing relative positioning. If no range is set, defaults
+       * to 1.
+       *
+       * Example Use:
+       * expect(panelTop).toBeApproximately(buttonTop, 1);
+       */
+      toBeApproximately: function() {
+        return {
+          compare: function(actual, expected, opt_offset) {
+            var offset = opt_offset || 1;
```
Comment (jelbourn) [inline src/components/panel/panel.spec.js]: optional: the traditional term for this is `epsilon`

File: src/components/panel/panel.spec.js [hunk 2]
```diff
@@ -31,6 +31,41 @@ describe('$mdPanel', function() {
...
+    jasmine.addMatchers({
+      /**
+       * Asserts that two values are within a range of each other. This is
+       * used for testing relative positioning. If no range is set, defaults
+       * to 1.
+       *
+       * Example Use:
+       * expect(panelTop).toBeApproximately(buttonTop, 1);
+       */
+      toBeApproximately: function() {
```
Comment (jelbourn) [inline src/components/panel/panel.spec.js]: How is this different from `toBeCloseTo` ?

**Comment by KarenParker** [comment]
LGTM

File: src/components/panel/panel.spec.js (hunk 2 above)
Comment (ErinCoughlan) [inline src/components/panel/panel.spec.js]: `toBeCloseTo` compares floats with a certain amount of decimal precision. I want to check that two numbers are within `x` of each other.

For instance, `expect(54.99).toBeCloseTo(55.01)` fails. With `toBeApproximately` this passes.

File: src/components/panel/panel.spec.js [hunk 3]
```diff
@@ -31,6 +31,41 @@ describe('$mdPanel', function() {
...
+          compare: function(actual, expected, opt_offset) {
+            var offset = opt_offset || 1;
+
+            var actualNumber = parseFloat(actual);
+            var expectedNumber = parseFloat(expected);
+
+            var rangeStart = expectedNumber - offset;
+            var rangeEnd = expectedNumber + offset;
+
+            var pass = actualNumber >= rangeStart && actualNumber <= rangeEnd;
```
Comment (jelbourn) [inline src/components/panel/panel.spec.js]: The typical way to do this is `Math.abs(num1 - num2) < epsilon`

**Comment by ErinCoughlan** [comment]
Thanks @jelbourn I was trying to figure out why this looked so strange yesterday.

I've fixed all the comments. PTAL.

**Comment by ErinCoughlan** [comment]
Merged.

**Pull request closed without merging**